examples/server.rs

```diff
@@ -18,6 +19,8 @@ impl LanguageServer for Backend {
     type DefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
     type TypeDefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
     type HighlightFuture = BoxFuture<Option<Vec<DocumentHighlight>>>;
+    type SignatureHelpFuture = BoxFuture<Option<SignatureHelp>>;
+    type GotoImplementationFuture = BoxFuture<Option<GotoImplementationResponse>>;
```

Comment:
I would generally like to keep the ordering of associated types and methods consistent in the order they are described in the [current supported LSP specification](https://microsoft.github.io/language-server-protocol/specifications/specification-3-15/).

In this case:

* `GotoImplementationFuture` should be moved to immediately below `TypeDefinitionFuture`.
* `SignatureHelpFuture` should be moved to immediately below `HoverFuture`.

Reply:
Additionally, to be consistent with the existing `TypeDefinitionFuture`, we should also rename this to be `ImplementationFuture`.